src/JsonApi/Serializer/ItemNormalizer.php

```diff
@@ -83,6 +83,10 @@ public function normalize(mixed $object, string $format = null, array $context =
             $resourceClass = $this->resourceClassResolver->getResourceClass($object, $context['resource_class'] ?? null);
         }
 
+        if (($operation = $context['operation'] ?? null) && method_exists($operation, 'getItemUriTemplate')) {
```

Comment:
These 3 lines are copied in multiple files, can't we move them in a trait? Something like ContextOperationTrait that helps to build the context properly from an operation (I'm almose sure we have lot of other context behaviors to move on it too).

Reply:
Not a huge fan of adding traits and a dependency between our normalizers.